enginetest/queries/index_queries.go

```diff
@@ -4110,6 +4110,51 @@ var IndexQueries = []ScriptTest{
 			},
 		},
 	},
+	{
+		// MySQL allows creating multiple indexes over the same set of columns. This isn't generally
+		// useful, but some customers need this support. For example, generated migration code from
+		// Django can create cases that require this: https://github.com/dolthub/dolt/issues/8254
+		Name: "multiple indexes over same set of columns",
+		SetUpScript: []string{
+			"CREATE TABLE `t0` (`id` char(32) NOT NULL PRIMARY KEY, `col1` varchar(255) NOT NULL, `col2` varchar(255) NOT NULL);",
+		},
+		Assertions: []ScriptTestAssertion{
+			// Add two indexes over the same column set to t0
+			{
+				Query:    "ALTER TABLE t0 ADD CONSTRAINT unique_1 UNIQUE(col1, col2);",
+				Expected: []sql.Row{{types.NewOkResult(0)}},
+			},
+			{
+				Query:    "ALTER TABLE t0 ADD CONSTRAINT unique_2 UNIQUE(col1, col2);",
+				Expected: []sql.Row{{types.NewOkResult(0)}},
+			},
+			{
+				Query: "SELECT kc.`constraint_name`, kc.`column_name`, kc.`referenced_table_name`, kc.`referenced_column_name` FROM information_schema.key_column_usage AS kc WHERE kc.table_schema = DATABASE() AND kc.table_name = 't0' ORDER BY kc.`ordinal_position`;",
+				Expected: []sql.Row{
+					{"PRIMARY", "id", nil, nil},
+					{"unique_1", "col1", nil, nil},
+					{"unique_2", "col1", nil, nil},
+					{"unique_1", "col2", nil, nil},
+					{"unique_2", "col2", nil, nil},
+				},
+			},
+			// Create a new table with two indexes over the same column set
+			{
+				Query:    "CREATE TABLE `t2` (`id` char(32) NOT NULL PRIMARY KEY, `col1` varchar(255) NOT NULL, `col2` varchar(255) NOT NULL, UNIQUE KEY unique_1(col1, col2), UNIQUE KEY unique_2(col1, col2));",
+				Expected: []sql.Row{{types.NewOkResult(0)}},
+			},
+			{
+				Query: "SELECT kc.`constraint_name`, kc.`column_name`, kc.`referenced_table_name`, kc.`referenced_column_name` FROM information_schema.key_column_usage AS kc WHERE kc.table_schema = DATABASE() AND kc.table_name = 't2' ORDER BY kc.`ordinal_position`;",
+				Expected: []sql.Row{
+					{"PRIMARY", "id", nil, nil},
+					{"unique_1", "col1", nil, nil},
+					{"unique_2", "col1", nil, nil},
+					{"unique_1", "col2", nil, nil},
+					{"unique_2", "col2", nil, nil},
+				},
+			},
```

Comment:
We should have some `SHOW CREATE TABLE` tests.

Reply:
Happy to add those. 👍  I originally preferred going to `information_schema` to validate the indexes are present, since `SHOW CREATE TABLE` output can be a little fragile, but it does totally make sense to test that `SHOW CREATE TABLE` works correctly when duplicate indexes are present. Good suggestion.